EIPS/eip-6357.md

```diff
@@ -0,0 +1,79 @@
+---
+eip: 6357
+title: Single-contract Multi-delegatecall
+description: Allows an EOA to call multiple functions of a smart contract in a single transaction
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/eip-6357-single-contract-multicall/12621
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-01-18
+---
+
+## Abstract
+
+This EIP standardizes an interface containing a single function, `multicall`, allowing EOAs to call multiple functions of a smart contract in a single transaction, and revert all calls if any call fails. 
+
+## Motivation
+
+Currently, in order to transfer several NFTs or SFTs, one needs to submit a number of transactions equal to the number of NFTs being tranferred. This wastes users' funds by requiring them to pay an [EIP-1559](./eip-1559.md) priority fee for every NFT they transfer.
```

Comment:
> This wastes users' funds by requiring them to pay an [EIP-1559](./eip-1559.md) priority fee for every NFT they transfer.

The priority fee is per gas used right? I think it would be better to complain about the 21k fee per transaction.